File: sliding-window/1838-Frequency-of-the-Most-Frequent-Element.go
func maxFrequency(nums []int, k int) int {
    sort.Ints(nums)
    total := 0
    frequency := 0

    left := 0
    for right, _ := range nums {
        total += nums[right]

        for nums[right] * (right - left + 1) - total > k {
            total -= nums[left]
            left++
        }

        frequency = max(frequency, right - left + 1)
    }

    return frequency
}

func max(a int, b int) int {
    if a > b {
        return a
    }
    return b
}

File: sliding-window/1838-Frequency-of-the-Most-Frequent-Element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        frequency = 0
        left, right = 0, 0
        total = 0
        while right < len(nums):
            total += nums[right]

            while nums[right] * (right - left + 1) - total > k:
                total -= nums[left]
                left += 1
            
            frequency = max(frequency, right - left + 1)
            right += 1
        return frequency